### PR TITLE
Add launchpad checklist for sites created in the hosting flow.

### PR DESCRIPTION
**Pull request by vykes-mac**
Fixes https://github.com/Automattic/dotcom-forge/issues/4012

## Proposed changes:

* Add launchpad checklist items for hosting trial sites.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply patch D126087-code
* checkout https://github.com/Automattic/wp-calypso/pull/83388 for calypso changes
* Create a new site from `/setup/new-hosted-site`
* Follow instructions below https://github.com/Automattic/jetpack/pull/33698#issuecomment-1771772622 or PCYsg-Osp-p2
* After setup jetpack beta tester ensure that the changes are also downloaded to your sandbox
* Verify the new checklist is shown and is ticked when completed

### CODE REVIEW

**Comment by mpkelly** [comment]
I tried to test this but can't get Jetpack Beta working. I get an error when uploading it to my site:
image

I tried to download the ZIP again, but it was the same. Don't have time to investigate further.

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php
Comment (zaguiini) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: Why do we need this `blog_id` checking?

Comment (vykes-mac) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: Not showing the new checklist for older sites/keeping the old checklist for those sites.

Comment (zaguiini) [reply projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: I see. It would be good if we could make this less magical (e.g., introduce a helper function, or at least extract the variable somewhere.

**Comment by vykes-mac** [comment]
@mpkelly  try uploading again

**Review by p-jackson — COMMENTED** [review]
I had issues testing the changes on a trial site because I wasn't able to upload the jetpack tester plugin. Could it be that plugin uploading isn't working for trial plans yet 😮 

But it's still testable on a regular paid business site.

Were you planning on running the `project-version.sh` script right before merging?

In my testing I found that the "Choose a theme" task was already selected by the time I saw the checklist. I wasn't expecting this. I'm sure under the hood a theme selection had already been made, but it wasn't part of the hosting onboarding steps.

CleanShot 2023-10-24 at 16 31 04@2x

The other unexpected thing is that "Give your site a name" is marked as incomplete, even though choosing a name was a required task during the hosting onboarding.

I notice here that the `write` intent intentionally hides the "Give your site a name" task because it's already a part of their onboarding flow: https://github.com/Automattic/jetpack/blob/89feea05e7bdcf2dcf6ea0d7f73f1ca1caf86757/projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php#L1532

So maybe we should just remove this task rather than figure out how to auto-complete it.

Other than the theme task everything else about the checklist works good. I left a comment on D126087-code too.

**Comment by p-jackson** [comment]
> I had issues testing the changes on a trial site because I wasn't able to upload the jetpack tester plugin. Could it be that plugin uploading isn't working for trial plans yet 😮

This will be fixed when https://github.com/Automattic/wpcomsh/pull/1531 deploys

**Comment by vykes-mac** [comment]
> The other unexpected thing is that "Give your site a name" is marked as incomplete, even though choosing a name was a required task during the hosting onboarding.

I think we are planning to remove the step to choose site name from the flow, at the moment we could update it when the name is generated, but I prefer to just remove it, It's not something that's important to showcasing the business features like the others.

> Were you planning on running the project-version.sh script right before merging?

Yeah

**Comment by vykes-mac** [comment]
@p-jackson The issue with the theme was that the hook get's executed with the initial site setup when the default theme is applied. I fix it by adding a check to see if the previous theme is the same as the updated theme. If it's the same then we do not mark the item as completed. 

can u check again, thanks!

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php
Comment (p-jackson) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php]: Typo here means they'll always be considered different themes.

```suggestion
	$is_same_theme = $new_theme['name'] === $old_theme['name'];
```

**Review by p-jackson — DISMISSED** [review]
Works just as expected 👍 All the tasks get completed when I expect now.

Although I don't quite know how the theme one is working correctly given the typo I point out below 🤔 It's working though.

I think this is good to go once the version scripts have been run and the typo has been fixed.

**Review by zaguiini — APPROVED** [review]

**Pull request merged**
